src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Queries/GetApprenticeships/GetApprenticeshipsQueryHandler.cs

```diff
@@ -70,68 +71,127 @@ public async Task<GetApprenticeshipsQueryResult> Handle(GetApprenticeshipsQuery
-                .Include(apprenticeship => apprenticeship.Cohort)
-                .Include(apprenticeship => apprenticeship.DataLockStatus)
-                .Include(apprenticeship => apprenticeship.ApprenticeshipUpdate)
-                .ToListAsync(cancellationToken);
-            
-            var apprenticeshipsWithoutAlerts = await _dbContext
-                .Apprenticeships
-                .Where(HasNoAlerts(providerId))
-                .Filter(filters)
-                .OrderBy(x => x.FirstName)
-                .ThenBy(x => x.LastName)
-                .ThenBy(x => x.Uln)
-                .ThenBy(x => x.Cohort.LegalEntityName)
-                .ThenBy(x => x.CourseName)
-                .ThenByDescending(x => x.StartDate)
-                .Include(apprenticeship => apprenticeship.Cohort)
-                .ToListAsync(cancellationToken);
+            var totalApprenticeshipsWithoutAlerts = await GetApprenticeshipsQuery(providerId, filters, false).CountAsync(cancellationToken);
+
+            var totalApprenticeshipsWithAlerts = await GetApprenticeshipsQuery(providerId, filters, true).CountAsync(cancellationToken);
 
-            var totalApprenticeshipsWithAlertsFound = apprenticeships.Count;
+            var skipCount = pageNumber > 0 ? (pageNumber - 1) * pageItemCount : 0;
 
-            List<Apprenticeship> combinedList;
+            var apprentices = new List<Apprenticeship>();
 
             if (reverseSort)
             {
-                combinedList = apprenticeshipsWithoutAlerts;
-                combinedList.AddRange(apprenticeships);
+                if (skipCount < totalApprenticeshipsWithoutAlerts)
+                {
+                    var apprenticeshipsWithoutAlerts =
+                        await GetApprenticeshipsWithoutAlerts(cancellationToken, providerId, skipCount, pageItemCount, filters);
+
+                    apprentices.AddRange(apprenticeshipsWithoutAlerts);
+                }
+
+                if (pageItemCount != 0 && apprentices.Count == pageItemCount)
+                {
+                    return new ApprenticeshipSearchResult
+                    {
+                        Apprenticeships = apprentices,
+                        TotalApprenticeshipsFound = totalApprenticeshipsWithoutAlerts + totalApprenticeshipsWithAlerts,
+                        TotalApprenticeshipsWithAlertsFound = totalApprenticeshipsWithAlerts
+                    };
+                }
+
+                skipCount -= totalApprenticeshipsWithoutAlerts;
+
+                var apprenticeshipsWithAlerts = await GetApprenticeshipsWithAlerts(cancellationToken, providerId, skipCount, pageItemCount, filters);
+
+                apprentices.AddRange(apprenticeshipsWithAlerts);
             }
             else
             {
-                combinedList = apprenticeships;
-                combinedList.AddRange(apprenticeshipsWithoutAlerts);
+                if (skipCount < totalApprenticeshipsWithAlerts)
```

Comment:
I feel like we should be able to remove some of this duplication from above - can we look at creating something for this?